Make it easier to create tensors manually

The easiest way I've found to create tensors so far is this:
```rust
let tensor: Tensor::<B, 2> = Tensor::from_data(Data::<f32 , 2>::from([[1.0f32, 2.0], [3.0, 4.0]]).convert());
```

This took quite a bit of digging to find out, and I'd expect this to be more straightforward.

### Suggestion 1 - Add a method directly to `Tensor`
```rust
// Note: I think it should be possible to infer the type on the LHS here (assuming the backend can be inferred from context)
let tensor: Tensor::<B, 2> = Tensor::from_array([[1.0f32, 2.0], [3.0, 4.0]]);
```

Not sure if `from_array` is the best name here.  We could also just use the `From` trait;

#### Implementation Details
This will require adding multiple impls like the following to Tensor:

```rust
impl<B: Backend>  Tensor<B, 1> {
    pub fn from_array<const D1: usize>(data: [B::Elem; D1]) -> Self {
        todo!()
    }
}

impl<B: Backend, >  Tensor<B, 2> {
    pub fn from_array<const D1: usize, const D2: usize>(data: [[B::Elem; D1]; D2]) -> Self {
        todo!()
    }
}

```

### Suggestion 2 - Add a macro
```rust
// Note: Again, the LHS type should be inferrable
let tensor: Tensor::<B, 2> = tensor![[1.0f32, 2.0], [3.0, 4.0]];
```

I just tried implementing suggestion 1, and got the following error from rust:
```
error[E0034]: multiple applicable items in scope
  --> examples/playground/src/main.rs:15:37
   |
15 |     let a: Tensor::<_, 1> = Tensor::from_array([1.0f32, 2.0]);
   |                                     ^^^^^^^^^^ multiple `from_array` found
   |
   = note: candidate #1 is defined in an impl for the type `Tensor<B, 1>`
   = note: candidate #2 is defined in an impl for the type `Tensor<B, 2>`
```

This is unfortunate, as I'd expect Rust to detect that only one of those `from_array` impls is applicable (both due to the array type and due to the expected tensor type)

--- a/burn-tensor/src/tensor/base.rs
+++ b/burn-tensor/src/tensor/base.rs
@@ -119,6 +119,23 @@ where
         Tensor::new(tensor)
     }
 
+    /// Create a tensor from floats (f32).
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use burn_tensor::backend::Backend;
+    /// use burn_tensor::Tensor;
+    ///
+    /// fn example<B: Backend>() {
+    ///     let _ = Tensor::<B, 1>::from_floats([1.0, 2.0]);
+    ///     let _ = Tensor::<B, 2>::from_floats([[1.0, 2.0], [3.0, 4.0]]);
+    /// }
+    /// ```
+    pub fn from_floats<A: Into<Data<f32, D>>>(floats: A) -> Self {
+        Self::from_data(floats.into().convert())
+    }
+
     /// Returns a new tensor with the same shape and device as the current tensor filled with zeros.
     pub fn zeros_like(&self) -> Self {
         Tensor::new(B::zeros(*self.shape(), self.device()))

